specs/experimental/contracts/governance/governance-token.md

```diff
@@ -0,0 +1,367 @@
+# GovernanceToken
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Assumptions](#assumptions)
+  - [aGT-001: Owner operates within governance constraints](#agt-001-owner-operates-within-governance-constraints)
+    - [Mitigations](#mitigations)
+  - [aGT-002: ERC20Votes total supply constraint](#agt-002-erc20votes-total-supply-constraint)
+    - [Mitigations](#mitigations-1)
+- [Invariants](#invariants)
+  - [iGT-001: Voting power reflects delegated balance](#igt-001-voting-power-reflects-delegated-balance)
+    - [Impact](#impact)
+  - [iGT-002: Delegation does not restrict transfers](#igt-002-delegation-does-not-restrict-transfers)
+    - [Impact](#impact-1)
+  - [iGT-003: Unauthorized minting is impossible](#igt-003-unauthorized-minting-is-impossible)
+    - [Impact](#impact-2)
+- [Function Specification](#function-specification)
+  - [mint](#mint)
+  - [burn](#burn)
+  - [burnFrom](#burnfrom)
+  - [delegate](#delegate)
+  - [delegateBySig](#delegatebysig)
+  - [getVotes](#getvotes)
+  - [getPastVotes](#getpastvotes)
+  - [getPastTotalSupply](#getpasttotalsupply)
+  - [delegates](#delegates)
+  - [transfer](#transfer)
+  - [transferFrom](#transferfrom)
+  - [approve](#approve)
+  - [permit](#permit)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The GovernanceToken is an ERC20 token that serves as the governance token for the OP Stack. It enables token holders to
+participate in governance through voting and delegation mechanisms while supporting controlled token inflation through
+owner-managed minting.
+
+## Definitions
+
+N/A
+
+## Assumptions
+
+### aGT-001: Owner operates within governance constraints
+
+The contract owner (MintManager) is trusted to mint tokens only in accordance with governance decisions and established
+protocol rules.
+
+#### Mitigations
+
+- Owner is expected to be a governance-controlled MintManager contract
+- MintManager enforces on-chain rate limits and caps on minting operations
+- Ownership transfers are transparent and subject to community oversight
+
+### aGT-002: ERC20Votes total supply constraint
+
+The total token supply never exceeds 2^208 - 1 tokens, as required by the inherited ERC20Votes implementation to prevent
+overflow in voting calculations.
+
+#### Mitigations
+
+- OpenZeppelin ERC20Votes implementation enforces this constraint in the _mint function
+- The constraint provides sufficient headroom for any realistic token supply
+
+## Invariants
```

Comment:
devinai Can you somehow reflect that "this is a standard ERC20 voting contract" is an invariant